tests/subsys/bluetooth/mesh/scheduler_model/src/main.c

```diff
@@ -0,0 +1,942 @@
+			.year = 10,
+			.month  = ANY_MONTH,
+			.day = 1,
+			.hour = 0,
+			.minute = 0,
+			.second = 30,
+			.day_of_week = ANY_DAY_OF_WEEK,
+			.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 0,
+			.scene_number = 1
+	};
+
+	action_put(&test_action);
+	measurement_start(30, 1);
+
+	struct tm expected = {
+		.tm_year = 110,
+		.tm_mon = 0,
+		.tm_mday = 1,
+		.tm_hour = 0,
+		.tm_min = 0,
+		.tm_sec = 30
+	};
+
+	expected_tm_check(&expected);
+}
+
+static void test_exact_time_second_ovflw(void)
+{ /* Tridonic use case */
+	const struct bt_mesh_schedule_entry test_action = {
+			.year = 10,
+			.month  = ANY_MONTH,
+			.day = 1,
+			.hour = 10,
+			.minute = 1,
+			.second = 5,
+			.day_of_week = ANY_DAY_OF_WEEK,
+			.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 0,
+			.scene_number = 1
+	};
+
+	start_tm.tm_hour = 10;
+	start_tm.tm_min = 0;
+	start_tm.tm_sec = 30;
+
+	action_put(&test_action);
+	measurement_start(35, 1);
+
+	struct tm expected = {
+		.tm_year = 110,
+		.tm_mon = 0,
+		.tm_mday = 1,
+		.tm_hour = 10,
+		.tm_min = 1,
+		.tm_sec = 5
+	};
+
+	expected_tm_check(&expected);
+}
+
+static void test_every_15_seconds(void)
+{
+	const struct bt_mesh_schedule_entry test_action = {
+			.year = BT_MESH_SCHEDULER_ANY_YEAR,
+			.month  = ANY_MONTH,
+			.day = BT_MESH_SCHEDULER_ANY_DAY,
+			.hour = 0,
+			.minute = BT_MESH_SCHEDULER_ANY_MINUTE,
+			.second = BT_MESH_SCHEDULER_EVERY_15_SECONDS,
+			.day_of_week = ANY_DAY_OF_WEEK,
+			.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 0,
+			.scene_number = 1
+	};
+
+	action_put(&test_action);
+	measurement_start(60, 4);
+
+	struct tm expected = {
+		.tm_year = 110,
+		.tm_mon = 0,
+		.tm_mday = 1,
+		.tm_hour = 0,
+		.tm_min = 1,
+		.tm_sec = 0
+	};
+
+	expected_tm_check(&expected);
+}
+
+static void test_every_20_seconds(void)
+{
+	const struct bt_mesh_schedule_entry test_action = {
+			.year = BT_MESH_SCHEDULER_ANY_YEAR,
+			.month  = ANY_MONTH,
+			.day = BT_MESH_SCHEDULER_ANY_DAY,
+			.hour = 0,
+			.minute = BT_MESH_SCHEDULER_ANY_MINUTE,
+			.second = BT_MESH_SCHEDULER_EVERY_20_SECONDS,
+			.day_of_week = ANY_DAY_OF_WEEK,
+			.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 0,
+			.scene_number = 1
+	};
+
+	action_put(&test_action);
+	measurement_start(60, 3);
+
+	struct tm expected = {
+		.tm_year = 110,
+		.tm_mon = 0,
+		.tm_mday = 1,
+		.tm_hour = 0,
+		.tm_min = 1,
+		.tm_sec = 0
+	};
+
+	expected_tm_check(&expected);
+}
+
+static void test_once_a_minute(void)
+{
+	const struct bt_mesh_schedule_entry test_action = {
+			.year = BT_MESH_SCHEDULER_ANY_YEAR,
+			.month  = ANY_MONTH,
+			.day = BT_MESH_SCHEDULER_ANY_DAY,
+			.hour = 0,
+			.minute = BT_MESH_SCHEDULER_ANY_MINUTE,
+			.second = BT_MESH_SCHEDULER_ONCE_A_MINUTE,
+			.day_of_week = ANY_DAY_OF_WEEK,
+			.action = BT_MESH_SCHEDULER_SCENE_RECALL,
+			.transition_time = 0,
+			.scene_number = 1
+	};
+
+	action_put(&test_action);
+	measurement_start(120, 1);
```

Comment:
Yep, @Andrewpini suggested a similar solution. I will start the action before 1 sec\min\hour before day expiration and do a series of events. It will give assurance that an event happens once in the time unit.
Btw. thanks for the suggestions it will really improve `once in a time unit` tests

Reply:
I think this is still not enough because the implementation of `bt_mesh_scene_srv_set` doesn't care about how many times a test allowed it to be called. And doesn't guarantee that all expected calls of `bt_mesh_scene_srv_set` were called in the appropriate interval. I think it would be good to add an assert if `gfire_cnt` is zero in `bt_mesh_scene_srv_set` and check that values in `fired_tm` are within the expected interval.